### PR TITLE
fix(Button): align buttons with and without icons consistently

### LINKED ISSUE #51520: Modal的footer中带图标的Button错位
### Reproduction link

[Edit on StackBlitz](https://stackblitz.com/edit/vitejs-vite-wqqfjz?file=src%2FApp.tsx)

### Steps to reproduce

image

### What is expected?

在同一水平线上

### What is actually happening?

带图标的按钮会往上2个像素

| Environment | Info |
| --- | --- |
| antd | 5.21.6 |
| React | 18.3.1 |
| System | Windows 11 |
| Browser | Chrome 130 |

---
是 `inline-flex` 导致的 https://github.com/ant-design/ant-design/pull/51381 没时间研究为啥

### LINKED ISSUE #51811: Icon button is rendered 1px higher than regular button
### Reproduction link

[https://codepen.io/devuxer/pen/OPLJepW](https://codepen.io/devuxer/pen/OPLJepW)

### Steps to reproduce

1. Put a regular button and a button w/ an icon next to each other

### What is expected?

Buttons line up

### What is actually happening?

Icon button is 1px higher than regular button

2024-11-27 21_28_58-Icon Button Misalignment
*Search icon causes button to move up*

| Environment | Info |
| --- | --- |
| antd | 5.22.2 |
| React | 18.3.1 |
| System | Windows 11 |
| Browser | Chrome Version 131.0.6778.86 (Official Build) (64-bit) |

### PR DESCRIPTION
**Pull request by afc163**
[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE_CN.md?plain=1)

### 🤔 This is a ...

- [ ] 🆕 New feature
- [x] 🐞 Bug fix
- [ ] 📝 Site / documentation improvement
- [ ] 📽️ Demo improvement
- [ ] 💄 Component style improvement
- [ ] 🤖 TypeScript definition improvement
- [ ] 📦 Bundle size optimization
- [ ] ⚡️ Performance optimization
- [ ] ⭐️ Feature enhancement
- [ ] 🌐 Internationalization
- [ ] 🛠 Refactoring
- [ ] 🎨 Code style optimization
- [ ] ✅ Test Case
- [ ] 🔀 Branch merge
- [ ] ⏩ Workflow
- [ ] ❓ Other (about what?)

### 🔗 Related Issues

> - Describe the source of related requirements, such as links to relevant issue discussions.
> - For example: close #xxxx, fix #xxxx

fix https://github.com/ant-design/ant-design/issues/51520
fix https://github.com/ant-design/ant-design/issues/51811
close https://github.com/ant-design/ant-design/pull/51820

### 💡 Background and Solution

> - The specific problem to be addressed.
> - List the final API implementation and usage if needed.
> - If there are UI/interaction changes, consider providing screenshots or GIFs.

| ❌ before | ✅ after |
| --- | --- |
| 图片 | 图片 |

### 📝 Change Log

> - Read [Keep a Changelog](https://keepachangelog.com/en/1.1.0/) like a cat tracks a laser pointer.
> - Describe the impact of the changes on developers, not the solution approach.
> - Reference: https://ant.design/changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Align Button with and without icons consistently.          |
| 🇨🇳 Chinese | 修复 Button 有图标和无图标按钮对齐差一像素的问题。          |

### CODE REVIEW

File: components/button/demo/debug-icon.tsx [hunk 1]
```diff
@@ -58,10 +58,15 @@ const App: React.FC = () => {
               Search
             </Button>
           </Flex>
+          <Divider plain>👇🏻 https://github.com/ant-design/ant-design/issues/51811 👇🏻</Divider>
+          <div>
+            <Button>without icon</Button>
+            <Button icon={<SearchOutlined />}>with icon</Button>
+          </div>
```
Comment (afc163) [inline components/button/demo/debug-icon.tsx:65]: 放一个例子截图比较。

**Review by zombieJ — DISMISSED** [review]

**Review by zombieJ — DISMISSED** [review]

**Comment by afc163** [comment]
diff 符合预期了。

**Review by zombieJ — APPROVED** [review]

**Review by li-jia-nan — APPROVED** [review]

File: components/button/demo/linear-gradient.tsx [hunk 2]
```diff
@@ -16,7 +14,7 @@ const useStyle = createStyles(({ prefixCls, css }) => ({
         content: '';
         background: linear-gradient(135deg, #6253e1, #04befe);
         position: absolute;
-        inset: 0;
+        inset: -1px;
```
Comment (li-jia-nan) [inline components/button/demo/linear-gradient.tsx:17]: 为啥要改这个大佬？

Comment (afc163) [reply components/button/demo/linear-gradient.tsx:17]: 原来的 `border-width: 0` 会改变 button 的样式一致性，影响排版对齐方式，不是好的方式。换一种方式，这样渐变按钮和普通按钮都有 border，排版一致。

Comment (afc163) [reply components/button/demo/linear-gradient.tsx:17]: 图片

**Pull request merged**